ServerNG.md

```diff
@@ -0,0 +1,155 @@
+Infinispan ServerNG
+====================
+
+Infinispan ServerNG is a reboot of Infinispan's server which addresses the following:
+
+* small codebase with as little duplication of already existing functionality (e.g. configuration)
+* embeddable: the server should allow for easy testability in both single and clustered configurations
+* RESTful admin capabilities
+* Logging using [JBoss Logging logmanager](https://github.com/jboss-logging/jboss-logmanager)
+* Security using [Elytron](https://github.com/wildfly-security/wildfly-elytron)
+
+# Layout
+
+The server is laid out as follows:
+
+* `/bin` scripts
+   * `server.sh` server startup shell script for Unix/Linux
+   * `server.ps1` server startup script for Windows Powershell 
+* `/lib` server jars
+   * `infinispan-server.jar` uber-jar of all dependencies required to run the server.
+* `/server` default server instance folder
+* `/server/log` log files
+* `/server/configuration` configuration files
+   * `infinispan.xml`
+   * keystores
+   * `logging.properties` for configuring logging
+   * User/groups property files (e.g. `mgmt-users.properties`, `mgmt-groups.properties`) 
+* `/server/data` data files organized by container name
+   * `default`
+      * `caches.xml` runtime cache configuration
+      * `___global.state` global state
+      * `mycache` cachestore data
+* `/server/lib` extension jars (custom filter, listeners, etc)
+
+# Paths
+
+The following is a list of _paths_ which matter to the server:
+
+* `infinispan.server.home` defaults to the directory which contains the server files.
+* `infinispan.server.root` defaults to the `server` directory under the `infinispan.server.home`
+* `infinispan.server.configuration` defaults to `infinispan.xml` and is located in the `configuration` folder under the `infinispan.server.root`
+
+# Command-line
+
+The server supports the following command-line arguments:
+
+* `-b`, `--bind-address=<address>`
+* `-c`, `--server-config=<config>`
+* `-o`, `--port-offset=<offset>`
+* `-s`, `--server-root=<path>`
+* `-v`, `--version`
+
+# Configuration
+
+The server configuration extends the standard Infinispan configuration adding server-specific elements:
+
+* `security` configures the available security realms which can be used by the endpoints.
+* `cache-container` multiple containers may be configured, distinguished by name.
+* `endpoints` lists the enabled endpoint connectors (hotrod, rest, ...).
+* `socket-bindings` lists the socket bindings.
+
+An example skeleton configuration file looks as follows:
+
+```
+<infinispan>
```

Comment:
Is this supposed to be the same schema as for embedded (will we just ignore the extras when used in embedded mode?), or something extending it?

Reply:
It is the same schema, which is already extensible through additional parsers (see the counters).
One main difference is allowing multiple cache-containers (identified by unique names).

Reply:
So what happens when you provide a config with these extra elements do embedded mode? Error/warning/ignored?